src/content/resources/glossary.md

```diff
@@ -279,3 +279,33 @@ If not, the pattern _refutes_, or denies, the match.
 Refutable patterns appear in [_matching contexts_][].
 
 [_matching contexts_]: /language/patterns#matching
+
+## Subclass
+
+A _subclass_ is a class that inherits from another class using the
+[`extends`](/language/extend) keyword.
+
+```dart
+class A extends B {} // A is a subclass of B; B is the superclass of A. 
+```
+
+Any subclass relation is also a [subtype](#subtype) relation,
+so you can use "subtype" to describe any subclass in Dart.
+However, the inverse is not true. Not all subtypes are subclasses.
+See the [subtype](#subtype) entry for more information.
+
+## Subtype
+
+A _subtype_ is a class that implements another class
+([`implements`](/language/classes#implicit-interfaces)),
+or inherits from another class ([`extends`](/language/extend)).
+
+```dart
+class A implements B {} // A is a subtype of B, but NOT a subclass of B.
+
+class C extends D {} // C is a subtype AND a subclass of D.
+```
+
+Everything in Dart is an object, so _subtype_ is also used to describe
+variable types in any context, not only explicitly declared class relationships.
+To learn more, see [The Dart type system](/language/type-system).
```

Comment:
It is true that we can have a subtype relationship that involves one or more type variables, e.g.,

```dart
class A<X, Y extends X> {
  // In here, `Y` is a subtype of `X`.
}
```

.. but it is confusing that 'variable types' is used to denote (apparently?) _anything_ which is not a class type?

Here are some types that aren't derived from classes:
- Record types.
- Function types.
- Type variables.
- Union types (`T?` or `FutureOr<T>`, where `T` can be any kind of type).
- Bottom types (types like `Never` and `X extends Never`).

Perhaps just call them 'other kinds of types' rather than 'variable types'?

Reply:
I mentioned this [earlier](https://github.com/dart-lang/site-www/pull/4958#discussion_r1218709423), but I think "everything is an object" is nebulous and has little to do with subtyping.

I wonder if it would help to have a language-agnostic explanation of subclassing vs. subtyping. A subclass inherits the implementation of its superclass, so subclassing is fundamentally about code/implementation reuse. Subtyping is about polymorphism: if `S` is a subtype of `T`, then something of type `S` can be used where something of type `T` is expected.

In Dart, a subclass relationship gives rise to a subtype relationship because a subclass also inherits its superclass's interface, and `extends SomeClass` implicitly states `implements SomeClass` as well. In principle, these concepts could have been decoupled at the cost of some ergonomics: an alternate-universe version of Dart could require you to say `extends SomeClass implements SomeClass` if you want both subtyping and subclassing.